addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1178 @@
+        """
+        #TOFIX: mail template should be used instead of fix body, subject text
+        details = []
+        result_type = opportunities._merge_get_result_type()
+        if result_type == 'lead':
+            merge_message = _('Merged leads')
+        else:
+            merge_message = _('Merged opportunities')
+        subject = [merge_message]
+        for opportunity in opportunities:
+            subject.append(opportunity.name)
+            title = "%s : %s" % (opportunity.lead_type == 'opportunity' and _('Merged opportunity') or _('Merged lead'), opportunity.name)
+            fields = list(CRM_LEAD_FIELDS_TO_MERGE)
+            details.append(opportunity._mail_body(fields, title=title))
+
+        # Chatter message's subject
+        subject = subject[0] + ": " + ", ".join(subject[1:])
+        details = "\n\n".join(details)
+        return self.message_post(body=details, subject=subject)
+
+    def _merge_opportunity_history(self, opportunities):
+        self.ensure_one()
+        for opportunity in opportunities:
+            for history in opportunity.message_ids:
+                history.write({
+                        'res_id': self.id,
+                        'subject': _("From %s : %s") % (opportunity.name, history.subject)
+                })
+        return True
+
+    def _merge_opportunity_attachments(self, opportunities):
+        # return attachments of opportunity
+        self.ensure_one()
+
+        def _get_attachments(opportunity_id):
+            attachment_ids = self.env['ir.attachment'].search([('res_model', '=', self._name), ('res_id', '=', opportunity_id)])
+            return attachment_ids
+
+        first_attachments = _get_attachments(self.id)
+        #counter of all attachments to move. Used to make sure the name is different for all attachments
+        count = 1
+        for opportunity in opportunities:
+            attachments = _get_attachments(opportunity.id)
+            for attachment in attachments:
+                values = {'res_id': self.id}
+                for attachment_in_first in first_attachments:
+                    if attachment.name == attachment_in_first.name:
+                        values['name'] = "%s (%s)" % (attachment.name, count,),
+                count += 1
+                attachment.write(values)
+        return True
+
+    def _create_lead_partner(self):
+        self.ensure_one()
+        contact_id = False
+        contact_name = self.contact_name or self.email_from and self.env['res.partner']._parse_partner_name(self.email_from)[0]
+        if self.partner_name:
+            partner_company = self._lead_create_contact(self.partner_name, True)
+        elif self.partner_id:
+            partner_company = self.partner_id
+        else:
+            partner_company = False
+
+        if contact_name:
+            contact_id = self._lead_create_contact(contact_name, False, partner_company and partner_company.id or False)
```

Comment:
1) partner_company and partner_company.id or False => partner_company.id  
2) self._lead_create_contact return recordset(s) so variable name should be contacts